src/main/scala/mesosphere/marathon/core/launcher/impl/InstanceOpFactoryImpl.scala

```diff
@@ -57,6 +59,14 @@ class InstanceOpFactoryImpl(
       case Instance(_, _, _, _, runSpec, _) =>
         throw new IllegalArgumentException(s"unsupported runSpec object ${runSpec}")
     }
+
+    result match {
+      case m: OfferMatchResult.Match => result
+      case _ if scheduledInstances.tail.nonEmpty =>
+        val tail = scheduledInstances.tail
+        matchOfferRequest(request.copy(scheduledInstances = NonEmptyIterable(tail.head, tail)))
+      case _ => result
+    }
```

Comment:
The implementation is not nice. We probably should just loop or use `collectFirst`

Reply:
not very performant as well probably if you have 1000 instances of the same runspec, then you're doing 999 matches that you did not have to do since it's the same runspec, the result is going to be the same

Reply:
Ah yeah. We could use something like `distinct` on the version.

Reply:
I'd keep it simple and predictable since we can have a special case due to existing reservations.

Reply:
> not very performant as well probably if you have 1000 instances of the same runspec

true, but all the other solutions we came up with (like having a `TaskLauncherActor` per app version) would be much more invasive. and having 1k instances seems to be very unrealistic so ¯\_(ツ)_/¯

Reply:
🙈 thanks for fixing the issue.